include/deal.II/distributed/cell_weights.h

```diff
@@ -68,9 +68,17 @@ namespace parallel
    *   hp_dof_handler.get_triangulation().signals.weight.connect(
    *     parallel::CellWeights<dim, spacedim>::make_weighting_callback(
    *       hp_dof_handler,
-   *       parallel::CellWeights<dim, spacedim>::ndofs_weighting({1, 1}));
+   *       parallel::CellWeights<dim, spacedim>::ndofs_weighting({1, 1})));
    * @endcode
    *
+   * If the weighting function only requires information about the finite
+   * element (for example about the number of degrees of freedom per cell), the
+   * weights can be computed in advance. Such a cache can be set up
+   * automatically by using the `enable_fe_cache` flag in the constructor or
+   * reinit() function, or by using the make_cached_weighting_callback()
+   * function. Note that no other cell-specific information can be used to build
+   * the cache in the current implementation.
```

Comment:
Perhaps write this as follows:
```suggestion
   * Oftentimes, the weighting function only requires information about the finite
   * element (for example about the number of degrees of freedom per cell) but not
   * the cell we are currently considering (even though the weighting function takes a
   * cell iterator as argument). In this case, it is wasteful to re-compute the weight on
   * ever cell we visit; instead, the weights can be computed for each finite element in
   * use on the DoFHandler in advance, and then cached. Such a cache can be set up
   * automatically by using the `enable_fe_cache` flag in the constructor or
   * reinit() function, or by using the make_cached_weighting_callback()
   * function. Of course, in this case no other cell-specific information can be used in the
   * computation of the weights.
```